config/middlewares/authUser.js

```diff
@@ -0,0 +1,26 @@
+require('dotenv').config();
+const mongoose = require('mongoose');
+const jwt = require('jsonwebtoken');
+
+const User = mongoose.model('User');
+
+module.exports = (req, res, next) => {
+  const token = req.header('Authorization');
+  if (!token) {
+    return res.status(401).json('Unauthorized: No access token provided');
+  }
+  jwt.verify(token, process.env.JWT_SECRET, (error, decoded) => {
+    if (error) {
+      return res.status(403).json('Invalid token');
+    }
+    User.findById(decoded.userId)
+      .then((user) => {
+        if (!user) {
+          return res.status(404)('User not found');
```

Comment:
This will error it should be res.status(404).send({message:'User not found'})

Reply:
got you

Reply:
done